com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -22,14 +23,195 @@ public LightweightRenderPipeline(LightweightRenderPipelineAsset asset)
 
 namespace UnityEngine.Rendering.Universal
 {
+    // A collection of profiling samplers, cache, helpers and convenience functions for Universal profiling.
+    // Keeps most of the profiling related things organized in one place for editability.
+    // Static global state, mostly read/sometimes write, non-thread safe (like most things in Universal).
+    //
+    // Example:
+    // // Cached CPU
+    // using var profScope = UniversalProfilingCache.GetCPUScope("example name");
+    // // Static GPU
+    // using( UniversalProfilingCache.GetGPUScope( cmd, UniversalProfilingScope.Pipeline.ExampleStaticSampler ) ) {}
+    internal static class UniversalProfilingCache
+    {
+        private static Dictionary<string, ProfilingSampler> m_StringSamplerCache = new Dictionary<string, ProfilingSampler>();
+        private static Dictionary<int, ProfilingSampler>    m_IntSamplerCache    = new Dictionary<int, ProfilingSampler>();
+        private static Dictionary<int, ProfilingSampler>    m_CameraSamplerCache = new Dictionary<int, ProfilingSampler>();
```

Comment:
Idea: have a look at FastPropertyName in C++. I think that way you could have a single dictionary <string, int>, while int is an index into a ProfilingSampler array.

 Then you could pre-populate / build at pipeline creation all static samplers in the array with indices that match the Enum.

Reply:
I reduced it to one string and one int map. I don't want to complicate this more at this point if it's not a problem.